### ISSUE TITLE
Support for Dgraph 1.1

### ISSUE THREAD

**Issue by ondrowan** [Swoorup/dgraph-rs#10]
Dgraph 1.1.0 [landed 5 days ago](https://github.com/dgraph-io/dgraph/releases/tag/v1.1.0) and it contains some breaking changes we need to address. I checked Go and JS clients and they both released new major version that supports it. I'm not 100% sure but I think we can mitigate this with feature flag. I'll try to look into it soon and see if it's gonna be as easy and elegant as I think.

**Comment by martinmr** [comment]
I think keeping two versions of the code is not that useful. Other than bug fixes, we don't have plans to change the clients for Dgraph 1.0.x. You can drop support for Dgraph 1.0.x in the next major version of your client and direct users of the 1.0.x series to keep using the old version of this client.

Keeping support for the two APIs in a single codebase would involve keeping two different versions of the generated protobuf files. IMO, it's a lot more trouble than it's worth.

**Comment by ondrowan** [comment]
The difference between Rust client and the official ones is that this one doesn't contain all the features (eg. JWT) just yet. It'd be a shame if people using Dgraph 1.0 wouldn't be able to use those, once they are implemented, just because we updated the codebase for Dgraph 1.1 meanwhile. Do you guys collect any usage data that could be helpful in this regard? Do people upgrade their servers regularly? What percentage uses 1.1 already?

**Comment by Pepeye** [comment]
As you mentioned, I think there are breaking changes. I got the following when running the simple example in your repo documentation/readme - note I removed all references to dob due to some other trait errors.

The error occurs on the following line

```rust
let assigned = txn.mutate(mutation).expect("failed to create data");
```

output:
```sh
Oct 29 09:26:25.855 INFO connect to dgraph via grpc at localhost:9080, scope: 1
Oct 29 09:26:25.856 INFO dropping all schema, scope: 1
Oct 29 09:26:25.902 INFO setup schema, scope: 1
Oct 29 09:26:25.966 INFO push data, scope: 1
thread 'main' panicked at 'failed to create data: GrpcError(RpcFailure(RpcStatus { status: Unimplemented, details: Some("unknown method Mutate for service api.Dgraph") }))', src/libcore/result.rs:1084:5
```

I think the Dgraph grpc api has changed as you alluded to. I've just come back to Dgraph after a little break and (I'm new to rust, previously used the go client) but I noticed also that it appears it supports types in the schema in version 1.1 but not 1.0.x. When do you think you'll update this to support 1.1? The following client appears to support 1.1 but requires rust nightly (rustc 1.39+).
scenario: 
* running dgraph in a local kubernetes cluster using the HA config
* rust stable 1.38

**Comment by ondrowan** [comment]
@Pepeye Since there's a real need for this, I'll at least look at the possibilities during the weekend and let you know.

**Comment by Pepeye** [comment]
@ondrowan Nice. I'm still learning/exploring rust but seriously want to stick with it. Compiler/borrow checker still winning the fight but I'm hanging in there.

Great job with this crate btw.

**Comment by insanitybit** [comment]
If it's helpful, I've started this work on my own dgraph library:
https://github.com/insanitybit/dgraph-rs

**Comment by ondrowan** [comment]
Dgraph 1.1+ support landed in version 0.3.0.

@Pepeye @d0nutptr @maxburke It'd be cool if you tried if this version fixes any issues you had.
